packages/probing-caller/src/utilities/kmsClientHandler.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  KMSClient,
+  SignCommand,
+  SigningAlgorithmSpec,
+  SignRequest,
+  SignCommandOutput,
+} from "@aws-sdk/client-kms";
+import { config } from "./config.js";
+import { v4 as uuidv4 } from "uuid";
+import { callerConstants } from "./constants.js";
+import { logger } from "pagopa-interop-probing-commons";
+import { buildJWTError } from "../model/domain/errors.js";
+
+interface Claims {
+  aud: string[];
+  sub: string;
+  nbf: number;
+  iss: string;
+  exp: number;
+  iat: number;
+  jti: string;
+}
+
+export const kmsClientBuilder = () => {
+  const kms: KMSClient = new KMSClient();
+
+  return {
+    async buildJWT(audience: string[]): Promise<string> {
+      try {
+        const token = createToken(audience);
+        const signReq: SignRequest = {
+          KeyId: config.jwtPayloadKidKms,
+          Message: Buffer.from(token),
+          SigningAlgorithm:
+            callerConstants.JWT_SIGNING_ALGORITHM as SigningAlgorithmSpec,
+        };
+
+        const signCommand = new SignCommand(signReq);
+        const signResult: SignCommandOutput = await kms.send(signCommand);
+        const signedTokenBuffer = signResult.Signature;
+        if (!signedTokenBuffer) {
+          throw new Error("Failed to generate signature.");
+        }
+
+        return removePadding(
+          `${token}.${Buffer.from(signedTokenBuffer).toString("base64")}`
+        );
```

Comment:
just to be more formal, padding removal is required for the signature
```suggestion
        return `${token}.${removePadding(Buffer.from(signedTokenBuffer).toString("base64"))}`;
```